packages/block-editor/src/store/reducer.js

```diff
@@ -1276,6 +1276,15 @@ function selectionHelper( state = {}, action ) {
  * @return {boolean} Updated state.
  */
 export function selection( state = {}, action ) {
+	if (
+		action.clientId ===
+		document.querySelector( '.wp-block-button' )?.dataset?.block
+	) {
+		action.clientId = document.querySelector(
```

Comment:
This really looks like [a hack](https://github.com/WordPress/gutenberg/pull/39463#issuecomment-1071021105) 😁  and not sure if there even is a proper implementation of this.